packages/carbon-web-components/src/components/side-panel/side-panel.ts

```diff
@@ -0,0 +1,1118 @@
+        await (this.constructor as typeof CDSSidePanel)._delay();
+        if (
+          !tryFocusElems(
+            this.querySelectorAll(selectorTabbableForSidePanel),
+            true
+          ) &&
+          relatedTarget !== this
+        ) {
+          this.focus();
+        }
+      }
+      // eslint-disable-next-line no-bitwise
+      else if (
+        relatedTarget === endSentinelNode ||
+        comparisonResult & FOLLOWING
+      ) {
+        await (this.constructor as typeof CDSSidePanel)._delay();
+        if (
+          !tryFocusElems(
+            this.querySelectorAll(selectorTabbableForSidePanel),
+            true
+          )
+        ) {
+          this.focus();
+        }
+      }
+    }
+  };
+
+  @HostListener('document:keydown')
+  // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
+  private _handleKeydown = ({ key, target }: KeyboardEvent) => {
+    if (key === 'Esc' || key === 'Escape') {
+      this._handleUserInitiatedClose(target);
+    }
+  };
+
+  private _reducedMotion = { matches: false };
+  // private _reducedMotion =
+  //   typeof window !== 'undefined' && window?.matchMedia
+  //     ? window.matchMedia('(prefers-reduced-motion: reduce)')
+  //     : { matches: true };
+
+  /**
+   * Handles `click` event on the side-panel container.
+   *
+   * @param event The event.
+   */
+  private _handleClickOnOverlay(event: MouseEvent) {
+    if (!this.preventCloseOnClickOutside) {
+      this._handleUserInitiatedClose(event.target);
+    }
+  }
+
+  /**
+   * Handles `click` event on the side-panel container.
+   *
+   * @param event The event.
+   */
+  private _handleCloseClick(event: MouseEvent) {
+    this._handleUserInitiatedClose(event.target);
+  }
+
+  /**
+   * Handles user-initiated close request of this side-panel.
+   *
+   * @param triggeredBy The element that triggered this close request.
+   */
+  private _handleUserInitiatedClose(triggeredBy: EventTarget | null) {
+    if (this.open) {
+      const init = {
+        bubbles: true,
+        cancelable: true,
+        composed: true,
+        detail: {
+          triggeredBy,
+        },
+      };
+      if (
+        this.dispatchEvent(
+          new CustomEvent(
+            (this.constructor as typeof CDSSidePanel).eventBeforeClose,
+            init
+          )
+        )
+      ) {
+        this.open = false;
+        this.dispatchEvent(
+          new CustomEvent(
+            (this.constructor as typeof CDSSidePanel).eventClose,
+            init
+          )
+        );
+      }
+    }
+  }
+
+  private _handleNavigateBack(triggeredBy: EventTarget | null) {
+    this.dispatchEvent(
+      new CustomEvent(
+        (this.constructor as typeof CDSSidePanel).eventNavigateBack,
+        {
+          composed: true,
+          detail: {
+            triggeredBy,
+          },
+        }
+      )
+    );
+  }
+
+  @state()
```

Comment:
can we place all the state declarations in the same section (maybe up top) so it's easier to reference and find?